crates/stackable-operator/src/commons/cache.rs

```diff
@@ -0,0 +1,86 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::time::Duration;
+
+/// [`TtlCache`] with sensible defaults for a user information cache
+pub type UserInformationCache = TtlCache<30, 10_000>;
+
+/// Least Recently Used (LRU) cache with per-entry time-to-live (TTL) value.
+///
+/// This struct has two const generics, so that different use-cases can have different default
+/// values:
+///
+/// * `D_TTL_SEC` is the default TTL (in seconds) the entries should have.
+/// * `D_MAX_ENTRIES` is the default for the maximum number of entries
+#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[schemars(
+    description = "Least Recently Used (LRU) cache with per-entry time-to-live (TTL) value."
+)]
+pub struct TtlCache<const D_TTL_SEC: u64, const D_MAX_ENTRIES: u32> {
```

Comment:
It's a neat idea, but it looks.. _very_ magical in the user code. I would consider instead pulling this out into a trait instead, something like:

```rs
pub trait TtlCacheDefaults {
    fn entry_time_to_live() -> Duration;
}
```

Reply:
I don't know how this can be achieved with a trait.

The following does not work:

```rust
#[serde(default = "impl TtlCacheDefaults::entry_time_to_live")]
pub entry_time_to_live: Duration,
```

Reply:
4bbdb42f94d5cd77726f3c6250512431c4f60dac

Reply:
Probably unpopular opinion: I find the const generics easier to read.
But I let you two decide ;)

Reply:
On the definition side I agree that the const generic is easier to read. On the *use side* I hate it when there are just two magic numbers with no indication of the meaning of either.

Reply:
I merged the trait branch into this one.